pkg/controllers/namespace/namespace_sync_controller.go

```diff
@@ -107,47 +108,59 @@ func (c *Controller) buildWorks(namespace *corev1.Namespace, clusters []clusterv
 		return err
 	}
 
-	for _, cluster := range clusters {
-		clonedNamespaced := namespaceObj.DeepCopy()
+	errChan := make(chan error, len(clusters))
```

Comment:
Do we really need to define the length of the channel? I think it's okay to use unbuffered channels.

Reply:
My consideration is that when there are too many ns, unbuffered pipe blocking may cause goroutine to hold too much

Reply:
ok.

In addition, perhaps we could add the appropriate namespace information when handling the error information, indicating which namespaces were in error.

Reply:
ok, wait a minute

Reply:
```go
fmt.Errorf("%v sync namespace %s failed, err: %v", cluster.GetName(), clonedNamespaced.GetName(), err)
```
What do you think

Reply:
Looks good to me~